syntax = "proto3";

package masterworker;

service MapReduceService {
  rpc PerformWork(WorkRequest) returns (WorkResponse) {}
}

message WorkRequest {
  string worker_type = 1;
  repeated file_info file_infos = 2;
  string user_id = 3;
  string output_directory = 4;
  int64 num_reducers = 5;
  repeated string file_names = 6;
  string worker_id = 7;
}

message WorkResponse {
  repeated string output_files = 2;
  bool success = 1;
}

message file_info {
  string file_name = 1;
  int64 start_bytes = 2;
  int64 end_bytes = 3;
}
